nbodykit/base/catalogmesh.py

```diff
@@ -122,38 +124,46 @@ def to_real_field(self):
             if len(Position) != 0:
 
                 # be sure to use the source to compute
-                position, weight, selection = self.source.compute(Position[s], Weight[s], Selection[s])
+                position, weight, value, selection = self.source.compute(Position[s], Weight[s], Mass[s], Selection[s])
             else:
                 # workaround a potential dask issue on empty dask arrays
                 position = numpy.empty((0, 3), dtype=Position.dtype)
                 weight = None
+                value = None
                 selection = None
 
             if weight is None:
                 weight = numpy.ones(len(position))
 
+            if value is None:
+                value = numpy.ones(len(position))
+
             if selection is not None:
                 position = position[selection]
                 weight   = weight[selection]
+                value     = value[selection]
+
+            Nlocal += value.sum()
```

Comment:
Shouldn't this be the sum of the Weight column? If Weight is like Mass? Or am I getting that wrong?